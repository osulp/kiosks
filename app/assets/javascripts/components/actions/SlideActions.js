export const SET_SLIDES = 'SET_SLIDES';
export const REFRESH_SLIDES = 'REFRESH_SLIDES';

export function setSlides(slides) {
  slides.push({id: 100, content: "absdg"});
  console.log(slides);
  return {
    type: SET_SLIDES,
    slides: slides
  };
}

export function refreshSlides(){
  // TODO: Ajax call to query the current set of slides?
}
