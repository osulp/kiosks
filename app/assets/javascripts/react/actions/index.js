export * from './kioskActions';
export * from './touchActions';
