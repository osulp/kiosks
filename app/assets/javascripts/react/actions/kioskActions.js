export const SET_KIOSK = 'SET_KIOSK';
export const setKiosk = (type, url) => {
  return {
    type: SET_KIOSK,
    kiosk: { type, url }
  };
};

export const SET_SLIDES = 'SET_SLIDES';
export const setSlides = (slides) => {
  return {
    type: SET_SLIDES,
    kiosk: {
      slides
    }
  };
};

export const SCROLL_TO_SLIDE = "SCROLL_TO_SLIDE";
export const scrollToSlide = (index) => {
  return {
    type: SCROLL_TO_SLIDE,
    index
  }
};

export const ADD_ERROR = 'ADD_ERROR';
export const addError = (error) => {
  return {
    type: ADD_ERROR,
    error
  };
};
