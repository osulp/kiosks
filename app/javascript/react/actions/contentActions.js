export const SET_CONTENT_ROOT_COMPONENT = "SET_CONTENT_ROOT_COMPONENT"
export const setContentRootComponent = root_component => {
  return {
    type: SET_CONTENT_ROOT_COMPONENT,
    root_component
  }
}
