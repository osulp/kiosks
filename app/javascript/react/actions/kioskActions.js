export const SET_KIOSK = "SET_KIOSK"
export const setKiosk = (
  type,
  url,
  map_default_floor_number,
  maps_base_url,
  name,
  kid
) => {
  return {
    type: SET_KIOSK,
    kiosk: { type, url, map_default_floor_number, maps_base_url, name, kid }
  }
}

export const SET_SLIDES = "SET_SLIDES"
export const setSlides = slides => {
  return {
    type: SET_SLIDES,
    kiosk: {
      slides
    }
  }
}

export const SET_RESTART_KIOSK = "SET_RESTART_KIOSK"
export const setRestartKiosk = restart_kiosk => {
  return {
    type: SET_RESTART_KIOSK,
    kiosk: {
      restart_kiosk
    }
  }
}

export const SCROLL_TO_SLIDE = "SCROLL_TO_SLIDE"
export const scrollToSlide = index => {
  return {
    type: SCROLL_TO_SLIDE,
    index
  }
}

export const ADD_ERROR = "ADD_ERROR"
export const addError = error => {
  return {
    type: ADD_ERROR,
    error
  }
}

export const SET_TITLE = "SET_TITLE"
export const setTitle = title => {
  return {
    type: SET_TITLE,
    title
  }
}

export const SET_GOOGLE_ANALYTICS = "SET_GOOGLE_ANALYTICS"
export const setGoogleAnalytics = google_analytics => {
  return {
    type: SET_GOOGLE_ANALYTICS,
    google_analytics
  }
}
