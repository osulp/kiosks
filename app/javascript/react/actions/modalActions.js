export const SET_MODAL_VISIBILITY = "SET_MODAL_VISIBILITY"
export const setModalVisibility = visible => {
  return {
    type: SET_MODAL_VISIBILITY,
    visible
  }
}

export const SET_MODAL_ROOT_COMPONENT = "SET_MODAL_ROOT_COMPONENT"
export const setModalRootComponent = root_component => {
  return {
    type: SET_MODAL_ROOT_COMPONENT,
    root_component
  }
}
