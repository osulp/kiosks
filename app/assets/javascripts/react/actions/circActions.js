import fetch from 'isomorphic-fetch';
import moment from 'moment';
import {addError, setSlides} from './kioskActions';

export const SET_ROOMS_AVAILABLE_COUNT = 'SET_ROOMS_AVAILABLE_COUNT';
export const setRoomsAvailableCount = (rooms_available_count, date) => {
  return {
    type: SET_ROOMS_AVAILABLE_COUNT,
    data: {
      rooms_available_count, date
    }
  };
};

export const FETCHING_ROOMS_AVAILABLE_COUNT = 'FETCHING_ROOMS_AVAILABLE_COUNT';
export const fetchingRoomsAvailableCount = () => {
  return {
    type: FETCHING_ROOMS_AVAILABLE_COUNT
  };
};

export const FETCHED_ROOMS_AVAILABLE_COUNT = 'FETCHED_ROOMS_AVAILABLE_COUNT';
export const fetchedRoomsAvailableCount = () => {
  return {
    type: FETCHED_ROOMS_AVAILABLE_COUNT
  };
};

export const fetchRoomsAvailableCount = (url, date) => {
  return (dispatch) => {
    dispatch(fetchingRoomsAvailableCount());
    // Default to today
    if (!date) {
      date = moment();
    }
    let parsed_date = moment(date).format("YYYYMMDDHHmmss");
    return fetch(`${url.replace("{date}", parsed_date)}`, {
      method: 'GET',
      headers: {
        'Accept': 'application/vnd.kiosks.v1',
        'Content-Type': 'application/json'
      }
    })
      .then(response => {
        switch (response.status) {
          case 200:
            return response.json();
          case 404:
            return {"error": "Rooms available not found for the selected date."};
          default:
            return {"error": "Error fetching rooms available, please notify the help desk."};
        }
      })
      .then(json => {
        dispatch(setRoomsAvailableCount(json, parsed_date));
        setTimeout(() => {
          dispatch(fetchedRoomsAvailableCount());
        }, 800);
      })
      .catch(err => {
        dispatch(addError({message: err.message, code: err.code}));
        dispatch(fetchedRoomsAvailableCount());
      });
  };
};

