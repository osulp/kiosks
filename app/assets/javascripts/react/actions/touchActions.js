import fetch from 'isomorphic-fetch';
import moment from 'moment';
import {addError, setSlides} from './kioskActions';

export const SET_MAPS = 'SET_MAPS';
export const setMaps = (maps) => {
  return {
    type: SET_MAPS,
    maps
  };
};

export const FETCHING_SLIDES = 'FETCHING_SLIDES';
export const fetchingSlides = () => {
  return {
    type: FETCHING_SLIDES
  };
};

export const FETCHED_SLIDES = 'FETCHED_SLIDES';
export const fetchedSlides = () => {
  return {
    type: FETCHED_SLIDES
  };
};

/**
 * A redux-thunk async action that fetches from the server and handles the return by dispatching a regular
 * action depending on success or error.
 * @param url - the url to the server for fetching slides
 * @returns {Redux Async Action} - see http://redux.js.org/docs/advanced/AsyncActions.html
 */
export const fetchSlides = (url) => {
  return (dispatch) => {
    dispatch(fetchingSlides());
    return fetch(`${url}.json`)
      .then(response => response.json())
      .then(json => {
        dispatch(setSlides(json.slides));
        // it's fast, so let the fetching_slides animation run for 800ms more. ;)
        setTimeout(() => {
          dispatch(fetchedSlides());
        }, 800);
      })
      .catch(err => {
        dispatch(addError({message: err.message, code: err.code}));
        dispatch(fetchedSlides());
      });
  };
};

export const SET_HOURS = 'SET_HOURS';
export const setHours = (hours) => {
  return {
    type: SET_HOURS,
    hours
  };
};

export const FETCHING_HOURS = 'FETCHING_HOURS';
export const fetchingHours = () => {
  return {
    type: FETCHING_HOURS
  };
};

export const FETCHED_HOURS = 'FETCHED_HOURS';
export const fetchedHours = () => {
  return {
    type: FETCHED_HOURS
  };
};

/**
 * A redux-thunk async action that fetches from the server and handles the return by dispatching a regular
 * action depending on success or error.
 * @param url - the url to the server for fetching slides
 * @returns {Redux Async Action} - see http://redux.js.org/docs/advanced/AsyncActions.html
 */
export const fetchHours = (url, dates) => {
  return (dispatch) => {
    dispatch(fetchingHours());
    // Default to this weeks dates if dates is undefined
    if (!dates) {
      dates = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"]
        .map(d => moment().day(d));
    }
    return fetch(url, {
      method: 'POST',
      headers: {
        'Accept': 'application/vnd.kiosks.v1',
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({dates:[...dates]})
    })
      .then(response => {
        switch(response.status) {
          case 200:
            return response.json();
          case 404:
            return {"error": "Hours not found for the selected date."};
          default:
            return {"error": "Error fetching hours, please notify the help desk."};
        }
      })
      .then(json => {
        dispatch(setHours(json));
        // it's fast, so let the fetching_hours animation run for 800ms more. ;)
        setTimeout(() => {
          dispatch(fetchedHours());
        }, 800);
      })
      .catch(err => {
        dispatch(addError({message: err.message, code: err.code}));
        dispatch(fetchedHours());
      });
  };
};
