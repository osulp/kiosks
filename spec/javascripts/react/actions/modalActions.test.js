import React from 'react';
import * as actionCreator from '../../../../app/assets/javascripts/react/actions/modalActions';

describe('Action::Modal', () => {
  describe('#setModalRootComponent()', () => {
    it('matches the snapshot', () => {
      let expected = React.createElement("H1", {}, "test");
      // execute
      let action = actionCreator.setModalRootComponent(expected);
      // verify
      expect(action).toMatchSnapshot();
    })
  });
  describe('#setModalVisibility()', () => {
    it('matches the snapshot', () => {
      // execute
      let action = actionCreator.setModalVisibility(true);
      // verify
      expect(action).toMatchSnapshot();
    })
  });
});