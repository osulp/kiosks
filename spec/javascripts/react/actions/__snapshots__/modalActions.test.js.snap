// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action::Modal #setModalRootComponent() matches the snapshot 1`] = `
Object {
  "root_component": <H1>
    test
  </H1>,
  "type": "SET_MODAL_ROOT_COMPONENT",
}
`;

exports[`Action::Modal #setModalVisibility() matches the snapshot 1`] = `
Object {
  "type": "SET_MODAL_VISIBILITY",
  "visible": true,
}
`;
